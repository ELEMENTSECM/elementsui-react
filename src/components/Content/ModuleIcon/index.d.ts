import * as React from 'react';

export type ModuleIconModuleId = 'rm' | 'mm' | 'sa' | 'eb';

export interface ModuleIconProps {
	/**
	 * HTML id tag of the root element
	 */
	id?: string;
	/**
	 * Module Id
	 */
	moduleId?: ModuleIconModuleId;
	/**
	 * Icon color
	 */
	color?: string;
	/**
	 * Icon size
	 */
	size?: number;
}

declare const ModuleIcon: React.SFC<ModuleIconProps>;

export default ModuleIcon;
