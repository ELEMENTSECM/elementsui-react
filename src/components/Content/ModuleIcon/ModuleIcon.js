import * as React from 'react';
import PropTypes from 'prop-types';
import { styles } from './ModuleIcon.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

export function ModuleIcon(props) {
	const { styles, moduleId, color, size } = props;
	const classNames = classNamesFunction()(styles, props);
	const icons = {
		rm: (
			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 179.88 164.89" fill={color} width={size}>
				<title>Record management module</title>
				<g id="Layer_2" data-name="Layer 2">
					<g id="Layer_1-2" data-name="Layer 1">
						<path d="M89.68,164.89H9.19c-6.39,0-9.19-2.76-9.19-9.07Q0,104.94,0,54.07C0,47.82,2.77,45,8.95,45h162c6.17,0,8.94,2.81,8.94,9.06q0,50.87,0,101.74c0,6.32-2.8,9.07-9.2,9.07Zm-.24-89.05.56.77c-.92-1.17-1.79-2.4-2.78-3.51a19.76,19.76,0,0,0-11.12-6.71c-7.32-1.4-13.48,1.15-18.69,6.08-6.75,6.38-10.86,17.21-4.77,27.19a54.25,54.25,0,0,0,6,8.27c6.71,7.4,13.72,14.52,20.43,21.92,3.9,4.3,7.44,8.92,11.27,13.55,8.48-12.21,19.23-22.33,29.26-33.1a68.2,68.2,0,0,0,6.42-7.95c6.67-9.76,5.89-20.16-2-29-7.5-8.39-18.87-9.73-27.6-3.19C94,72,91.76,73.94,89.44,75.84Z" />
						<path d="M172.38,37.31H7.66c0-5.45-.28-10.85.1-16.2.25-3.55,3.53-6,7.17-6.08a61.21,61.21,0,0,1,7.49,0c2.11.19,3-.65,3.45-2.56a45.94,45.94,0,0,1,1.56-4.74A10.92,10.92,0,0,1,38.18,0q18,0,36,0c5.17,0,9,2.81,10.77,7.73.84,2.34,1.06,5.68,2.72,6.75,1.92,1.22,5.12.5,7.77.5q33.87,0,67.75,0c6.4,0,9.17,2.74,9.19,9.07C172.38,28.41,172.38,32.73,172.38,37.31Z" />
					</g>
				</g>
			</svg>
		),
		sa: (
			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 252.09 198.97" fill={color} height={size}>
				<title>System administration module</title>
				<g id="Layer_2" data-name="Layer 2">
					<g id="Layer_1-2" data-name="Layer 1">
						<path d="M34.35,18.17c6.18,3.7,12.36,7.3,18.43,11.08a4.26,4.26,0,0,0,4.72.23A94.28,94.28,0,0,1,69,24.78a4.19,4.19,0,0,0,3.18-3.49C73.69,15,75.34,8.69,76.84,2.38,77.22.79,77.81,0,79.64,0c6.06.1,12.14.13,18.2,0,2.18-.06,2.8.91,3.24,2.78,1.46,6.32,3.12,12.59,4.6,18.9a3.71,3.71,0,0,0,2.78,3c4.14,1.46,8.22,3.16,12.22,5a3.53,3.53,0,0,0,3.85-.1c6.23-3.85,12.54-7.56,18.92-11.39A79.54,79.54,0,0,1,159,33.61a2.74,2.74,0,0,1-.23,2.35C155.41,41.71,152,47.44,148.5,53.1a3.9,3.9,0,0,0-.22,4.3c1.84,3.9,3.44,7.92,4.92,12a3.69,3.69,0,0,0,3,2.81c6.48,1.49,12.91,3.23,19.38,4.76,1.4.33,2.28.75,2.27,2.35q-.06,9.72,0,19.43c0,1.48-.82,1.94-2.08,2.24-6.63,1.61-13.26,3.21-19.84,5a4.39,4.39,0,0,0-2.41,2.26c-1.88,4.16-3.46,8.45-5.33,12.61a3.42,3.42,0,0,0,.17,3.65c3.53,5.74,6.92,11.57,10.45,17.31.77,1.25.9,2.14-.21,3.23q-6.84,6.74-13.57,13.58c-1.12,1.14-2,.92-3.26.17-5.68-3.48-11.45-6.81-17.09-10.34-1.5-.94-2.58-.74-4.08-.06-4,1.82-8.07,3.54-12.22,5a3.47,3.47,0,0,0-2.61,2.83c-1.53,6.39-3.18,12.74-4.68,19.14-.42,1.78-1.12,2.72-3.13,2.69-6.06-.1-12.13-.07-18.19,0-1.71,0-2.48-.69-2.87-2.36-1.52-6.48-3.2-12.91-4.73-19.39a3.66,3.66,0,0,0-2.81-2.94c-4.22-1.48-8.34-3.28-12.43-5.13a3.17,3.17,0,0,0-3.42.11c-5.74,3.53-11.57,6.92-17.31,10.46-1.34.83-2.3.92-3.46-.25-4.38-4.44-8.77-8.86-13.24-13.21-1.38-1.34-1.09-2.41-.2-3.84,3.47-5.59,6.78-11.27,10.26-16.85a3.37,3.37,0,0,0,.16-3.64c-1.9-4.24-3.53-8.6-5.42-12.85A4.09,4.09,0,0,0,22,106c-6.58-1.77-13.2-3.4-19.84-5C.76,100.66,0,100.14,0,98.57c.05-6.48,0-13,0-19.43C0,77.83.65,77.32,1.92,77c6.63-1.58,13.23-3.32,19.86-4.88a3.39,3.39,0,0,0,2.68-2.49c1.61-4.27,3.38-8.48,5.25-12.65a3.19,3.19,0,0,0-.12-3.41Q24.24,44.86,19.09,36a2.91,2.91,0,0,1-.19-2.53A79.57,79.57,0,0,1,34.35,18.17Zm85.92,70.91a31.37,31.37,0,1,0-31.53,31.28A31.38,31.38,0,0,0,120.27,89.09Z" />
						<path d="M182.36,190.28l-7.54-7.48c2.12-3.57,3.9-6.38,5.47-9.3a4.63,4.63,0,0,0,.14-2.69c-.11-.71-.8-1.33-.9-2-.56-4-3.27-5.37-6.84-5.78a11.16,11.16,0,0,1-1.45-.38c-5.05-1.29-5.05-1.29-5.07-6.56,0-.17,0-.33,0-.5,0-4.85,0-4.85,4.78-6l.24-.06c7.12-1.81,7.15-1.8,9.48-8.88a3.17,3.17,0,0,0-.41-2.23c-1.59-2.82-3.29-5.59-5.36-9.05l7.71-7.49c3.25,2,6,3.79,8.94,5.4a4.48,4.48,0,0,0,2.66.16c.7-.11,1.31-.82,2-.93,4.15-.68,5.3-3.54,6-7.24,1.28-6.35,1.47-6.31,8-6.07,4.34.16,4.34.16,5.36,4.34,1.81,7.43,1.8,7.46,8.87,10.1a3,3,0,0,0,2.22-.33c2.88-1.65,5.7-3.42,9.18-5.53l7.49,7.53c-2,3.39-3.79,6.19-5.34,9.1-.41.78-.17,1.92-.18,2.9,0,.22.26.44.3.68.93,5.42,4.8,7,9.61,7.75,4.28.7,4.24,1,4.3,5.34.09,6.38.14,6.63-5.94,7.8-3.94.76-6.44,2.19-7.69,6.32-.71,2.36-1.07,4,.42,6.08s2.77,4.53,4.53,7.49l-7.71,7.38c-3.16-1.87-6-3.64-8.87-5.19a5.13,5.13,0,0,0-2.89-.15c-.7.1-1.32.81-2,.91-4.16.58-5.42,3.45-5.8,7.08a6.88,6.88,0,0,1-.24,1c-1.36,5.3-1.36,5.3-6.92,5.26-5.14,0-5.14,0-6.39-5-1.79-7.05-1.78-7.08-8.79-9.54a3.08,3.08,0,0,0-2.22.37C188.69,186.42,185.93,188.13,182.36,190.28ZM209,139.9a16.1,16.1,0,1,0,16.22,15.83A16.08,16.08,0,0,0,209,139.9Z" />
					</g>
				</g>
			</svg>
		),
		mm: (
			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 262.46 164.76" fill={color} height={size}>
				<title>Meeting module</title>
				<g id="Layer_2" data-name="Layer 2">
					<g id="Layer_1-2" data-name="Layer 1">
						<path d="M131.25,164.76H49.52c-4,0-4.5-.48-4.46-4.42.08-7.07.14-14.15.36-21.22a10.58,10.58,0,0,1,6-9.23q25.42-13.84,51-27.45c1.92-1,2.83-1.91,2.77-4.24-.18-7.74,0-15.49-.13-23.24a5.21,5.21,0,0,0-1.45-3.33A18.36,18.36,0,0,1,97.7,58.24c-.12-10.07-.34-20.18.42-30.2C99.25,13,110.43,2.2,125.39.4c8.84-1.07,17.31-.11,25,4.57A27.7,27.7,0,0,1,164.2,27c.88,11.16.44,22.44.2,33.66-.09,4.37-2.48,8-5.68,11.06a4.55,4.55,0,0,0-1.23,2.95c-.1,8.08,0,16.16-.1,24.24,0,1.72.62,2.42,2.11,3.1,17.69,8,34.5,17.65,51.21,27.51,4.58,2.7,6.59,6.67,6.66,11.76.09,6.75.08,13.5,0,20.24,0,2.54-.89,3.19-3.94,3.19H131.25Z" />
						<path d="M88.8,96.11l-2.63,1.46L48.07,118.4c-6.68,3.65-11.85,8.58-13.58,16.45H8.88c-1.92,0-3.84-.14-5.74,0C.6,135.13-.06,134,0,131.68c.13-3.49.17-7,0-10.49A10.27,10.27,0,0,1,5.46,111c8.51-4.88,17-9.84,25.62-14.49,4-2.16,8.39-3.62,12.54-5.52A2.67,2.67,0,0,0,45,89.12c.11-5.58.13-11.16,0-16.74a3.2,3.2,0,0,0-1.77-2.13c-3-1.36-5.41-3.35-5.43-6.81-.05-7.73-.64-15.6.52-23.17C40.35,26.9,50.09,21.37,63,22.73,74.63,24,81,30.5,82,42.18c.61,6.69.3,13.48.29,20.22a7.54,7.54,0,0,1-5.45,7.54c-1.59.57-2,1.53-2,3.19.12,5.16.1,10.33,0,15.49a2.36,2.36,0,0,0,1.89,2.67C80.67,92.71,84.47,94.35,88.8,96.11Z" />
						<path d="M173.91,96c4-1.62,7.64-3.21,11.38-4.54,1.65-.59,2.37-1.27,2.31-3.14-.15-5.08-.11-10.16,0-15.24a2.61,2.61,0,0,0-1.88-3c-2.94-1.14-5.37-3.06-5.35-6.52,0-8.05-.52-16.23.67-24.13,2-13.43,13.06-18,24.53-16.7,11.67,1.36,18,7.81,19,19.54.59,6.61.29,13.31.3,20a7.51,7.51,0,0,1-5.32,7.63c-1.7.61-2.18,1.6-2.14,3.35.11,5.16.09,10.33,0,15.49a2.2,2.2,0,0,0,1.75,2.5c13.71,4.95,26.1,12.44,38.37,20.13a10,10,0,0,1,4.93,9.45c-.12,3.74-.08,7.5,0,11.24,0,2-.58,2.88-2.72,2.86-10-.07-20,0-30-.08-.71,0-1.89-.59-2-1.12-2-7.41-7.28-11.92-13.66-15.45L175.38,96.88C175,96.68,174.67,96.45,173.91,96Z" />
					</g>
				</g>
			</svg>
		),
		eb: (
			<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 179.14 165.04" fill={color} width={size}>
				<title>eBuildingCase module</title>
				<g id="Layer_2" data-name="Layer 2">
					<g id="Layer_1-2" data-name="Layer 1">
						<path d="M179.14,164.89h-37.5v-2.68c0-8,0-16,0-24,0-2.34-.83-3.27-3.23-3.26q-15.62.12-31.24,0c-2.28,0-3.1.92-3.09,3.14.06,8.83,0,17.66,0,26.69H83.31V129c6.71.31,12.63-.82,14-9h4.56a2,2,0,0,0,2.3-2.32c0-3.42,0-6.83,0-10.25,0-1.8-1-2.46-2.74-2.44-4.16,0-8.33.08-12.49,0a4.42,4.42,0,0,1-2.78-1.16Q76.86,94.32,67.8,84.66a4.54,4.54,0,0,1-1.11-2.83q-.09-38.61,0-77.22C66.65,1.42,68,0,71.19,0q51.73,0,103.46,0c3.2,0,4.48,1.46,4.48,4.92q0,38.36,0,76.72,0,40,0,80Zm-26-59.89c-2.91,0-5.83,0-8.74,0-1.89,0-2.75.83-2.73,2.7q0,4.87,0,9.74c0,1.77.78,2.59,2.56,2.58q8.74,0,17.48,0c1.67,0,2.47-.74,2.46-2.39,0-3.33,0-6.66,0-10,0-1.9-.93-2.68-2.79-2.64C158.6,105,155.85,105,153.11,105Zm-30.33,15c2.83,0,5.66-.05,8.49,0,2,.05,2.94-.71,2.89-2.77-.07-3.16,0-6.32,0-9.48,0-2-.93-2.81-2.91-2.79q-8.36.08-16.72,0c-2,0-2.91.8-2.89,2.81,0,3.16.06,6.32,0,9.48,0,2.08,1,2.79,2.92,2.74C117.28,119.95,120,120,122.78,120Zm30.15-22.5c2.83,0,5.66-.05,8.49,0,1.9,0,2.76-.73,2.73-2.65,0-3.25,0-6.49,0-9.74,0-1.9-.92-2.66-2.79-2.65q-8.49.06-17,0c-1.89,0-2.75.82-2.74,2.7,0,3.25,0,6.49,0,9.74,0,1.77.77,2.61,2.56,2.59C147.1,97.47,150,97.5,152.93,97.5Zm-30.14,0c2.83,0,5.66-.05,8.49,0,2,.05,2.93-.72,2.88-2.78q-.1-4.62,0-9.23c0-2.07-.75-3.05-2.93-3-5.57.07-11.15.06-16.72,0-2,0-2.9.82-2.88,2.82q.07,4.74,0,9.48c0,1.88.76,2.76,2.67,2.73C117.13,97.46,120,97.5,122.79,97.5Zm-29.65-15c-2.91,0-5.82,0-8.74,0-1.88,0-2.78.78-2.77,2.67,0,3.24,0,6.49,0,9.73,0,1.94.92,2.64,2.77,2.63,5.66,0,11.32-.05,17,0,2.06,0,2.83-.9,2.78-2.88q-.1-4.62,0-9.23c0-2.15-.92-3-3-2.92C98.46,82.57,95.8,82.5,93.14,82.5ZM152.91,75c2.82,0,5.65-.06,8.47,0,2.06.06,2.83-.85,2.78-2.84-.07-3.15-.06-6.31,0-9.47,0-1.88-.76-2.74-2.67-2.72q-8.59.06-17.19,0c-1.92,0-2.69.86-2.67,2.73,0,3.24,0,6.48,0,9.72,0,1.76.77,2.6,2.55,2.58C147.09,75,150,75,152.91,75Zm-30,0c2.83,0,5.66,0,8.5,0,1.86,0,2.79-.7,2.76-2.64q-.07-4.87,0-9.75c0-1.94-.91-2.66-2.77-2.64q-8.5.06-17,0c-1.86,0-2.79.71-2.76,2.64q.07,4.87,0,9.75c0,1.94.9,2.68,2.76,2.64C117.22,75,120.06,75,122.89,75ZM93.12,60c-3,0-6,0-9,0-1.64,0-2.48.73-2.48,2.4q0,5.12,0,10.25c0,1.69.9,2.36,2.52,2.36q8.75,0,17.49,0c1.82,0,2.51-.92,2.49-2.65,0-3.25,0-6.5,0-9.75,0-2-.94-2.66-2.79-2.62C98.61,60,95.86,60,93.12,60Zm59.73-7.5c2.92,0,5.83,0,8.75,0,1.82,0,2.57-.82,2.55-2.57,0-3.33,0-6.67,0-10a2.14,2.14,0,0,0-2.44-2.44c-5.83,0-11.67,0-17.5,0-1.79,0-2.58.81-2.56,2.58,0,3.25,0,6.5,0,9.75,0,1.91.82,2.72,2.72,2.69C147.18,52.46,150,52.5,152.84,52.5Zm-29.92,0c2.92,0,5.83,0,8.75,0a2.13,2.13,0,0,0,2.47-2.42c0-3.33,0-6.66,0-10,0-1.77-.76-2.6-2.55-2.6q-8.75,0-17.49,0c-1.65,0-2.47.76-2.46,2.42,0,3.33,0,6.66,0,10,0,2,1,2.64,2.8,2.6C117.26,52.46,120.09,52.5,122.93,52.5Zm-30,0c2.83,0,5.66,0,8.5,0,1.86,0,2.8-.68,2.78-2.63q-.07-4.87,0-9.75c0-1.93-.89-2.67-2.75-2.65q-8.5.06-17,0c-1.86,0-2.8.69-2.78,2.63q.07,4.87,0,9.75c0,1.93.88,2.69,2.75,2.65C87.21,52.46,90,52.5,92.88,52.5Zm60-22.5c2.83,0,5.66-.05,8.49,0,1.88,0,2.79-.69,2.76-2.62,0-3.25,0-6.49,0-9.74,0-1.89-.88-2.69-2.76-2.68q-8.49.06-17,0c-1.88,0-2.78.78-2.77,2.67,0,3.25,0,6.49,0,9.74,0,1.95.93,2.66,2.78,2.62C147.24,30,150.07,30,152.9,30ZM123,15c-2.83,0-5.66.05-8.49,0-2,0-2.92.8-2.89,2.81,0,3.16,0,6.32,0,9.48,0,1.87.74,2.75,2.66,2.74q8.61-.06,17.22,0c1.94,0,2.67-.9,2.64-2.76-.05-3.16,0-6.32,0-9.48,0-2-.93-2.84-2.91-2.78C128.49,15.05,125.75,15,123,15ZM92.8,30c2.91,0,5.82,0,8.74,0,1.76,0,2.63-.73,2.61-2.53q0-4.87,0-9.73c0-1.84-.75-2.77-2.68-2.76q-8.61.05-17.22,0c-1.77,0-2.61.82-2.61,2.57,0,3.33,0,6.66,0,10,0,1.83.95,2.49,2.67,2.46C87.14,30,90,30,92.8,30Z" />
						<path d="M0,119.32c.57-.65,1.11-1.34,1.71-2L41.79,76.5c2.66-2.71,3-2.72,5.56,0Q67,96.92,86.62,117.4c.59.62,1.1,1.32,2.08,2.49H74.14v3q0,19.37,0,38.74c0,3.12-.3,3.41-3.46,3.42-5.33,0-10.66-.06-16,0-2.28,0-3.1-.87-3.08-3.11.08-6.39,0-12.78,0-19.29H36.75c0,.79-.11,1.58-.11,2.38,0,5.58-.07,11.16,0,16.74,0,2.32-.72,3.34-3.16,3.29q-8.12-.17-16.24,0c-2.44.05-3.17-1-3.16-3.29.07-12.91,0-25.82,0-38.74,0-.9,0-1.81,0-3H.37Z" />
						<path d="M29.14,67.68c0,2.73.06,5.3,0,7.86a3.12,3.12,0,0,1-.83,1.93C23.65,82,19,86.51,14.36,91V67.68Z" />
					</g>
				</g>
			</svg>
		)
	};

	return <div className={classNames.root}>{icons[moduleId]}</div>;
}

ModuleIcon.propTypes = {
	/** Module Id */
	moduleId: PropTypes.string,
	/** Icon color */
	color: PropTypes.string,
	/** Icon size */
	size: PropTypes.number
};

export default styled(customizable('ModuleIcon', ['theme'])(ModuleIcon), styles);
