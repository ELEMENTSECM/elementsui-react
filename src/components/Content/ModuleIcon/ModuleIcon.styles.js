export const styles = props => {
	const { className, size } = props;

	return {
		root: [
			'ModuleIcon',
			{
				height: size
			},
			className
		]
	};
};
