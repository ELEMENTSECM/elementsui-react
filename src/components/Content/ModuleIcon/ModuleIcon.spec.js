import * as React from 'react';
import { shallow } from 'enzyme';
import ModuleIcon from './index';

describe('ModuleIcon', () => {
	test('displays correctly', () => {
		const wrapper = shallow(<ModuleIcon moduleId="rm" color="red" size={200} />);
		expect(wrapper).toMatchSnapshot();
	});
});
