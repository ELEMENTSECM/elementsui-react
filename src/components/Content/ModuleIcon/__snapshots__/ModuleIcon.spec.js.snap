// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleIcon displays correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledCustomizedModuleIcon
    color="red"
    moduleId="rm"
    size={200}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "color": "red",
      "styles": [Function],
      "moduleId": "rm",
      "size": 200,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "red",
        "styles": [Function],
        "moduleId": "rm",
        "size": 200,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
